java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -308,24 +315,24 @@ abstract class AbstractArcHost(vararg initialParticles: ParticleRegistration) :
     ) = when (handleSpec.type) {
         is SingletonType<*> ->
             entityHandleManager.createSingletonHandle(
-                holder,
+                holder.getEntitySpec(handleName),
                 handleName,
                 handleSpec.storageKey,
                 handleSpec.type.toSchema()
             )
         is CollectionType<*> ->
             entityHandleManager.createSetHandle(
-                holder,
+                holder.getEntitySpec(handleName),
                 handleName,
                 handleSpec.storageKey,
                 handleSpec.type.toSchema()
             )
         else -> throw IllegalArgumentException("Unknown type ${handleSpec.type}")
-    }
+    }.also { holder.setHandle(handleName, it) }
 
     override suspend fun stopArc(partition: Plan.Partition) {
         val arcId = partition.arcId
-        val context = lookupOrCreateArcHostContext(partition)
+        lookupOrCreateArcHostContext(partition)
```

Comment:
the lookupOrCreate line can be deleted. stopArc() can't possibly create a context, one must already exist, which is why the line below looks up the context via runningArcs[partition.arcId] and only executes if it's non null.

My guess is I was refactoring and that no-op line got left as I was deleting code.

Reply:
Done